specs/schemes/exact/scheme_exact_algo.md

```diff
@@ -0,0 +1,173 @@
+
+Example:
+
+```json
+{
+  "paymentGroup": [
+    {
+      "signature": "wVGxbZwPd2c8zKX4RSE4muFRBOHe/nRq5VT6iMeMksi2agfOCTs+VzOGUPJ+bTT5T21EuYjoBKKDX2+/NZG8Cg==",
+      "txn": {
+        "gen": "mainnet-v1.0",
+        "gh": "SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=",
+        "asnd": "PPNHVQJGE6ZMEWPR35UHLUYMCCZV6VNTHLJMZDVCONXKUPV47K4WGQJWGI",
+        "arcv": "ACTSMNPJPS5JMZJJ5G73JOXUUMWXXDGS7TMOER3M4W7BC54ERSZ4KPSMHM",
+        "type": "axfer",
+        "fee": "0",
+        "fv": "6631154",
+        "lv": "6632154",
+        "xaid": "168103"
+      }
+    },
+    {
+      "signature": "6iYMGg8cMw0VYYCfmwru8lixfrVyi1LFXm9H/M0MGf0P4vjK6KGsiCUvsocWUvMo8yTMWQ0hFcP86sRoyGlIAg==",
+      "txn": {
+        "gen": "mainnet-v1.0",
+        "gh": "SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=",
+        "fee": "2000",
+        "fv": "6631154",
+        "lv": "6632154",
+        "snd": "LOXIQKHRCEDEMN5MKBQ32Y54J7H6JKBTEURQL4S65K44MTWN6UMTU2WNBM",
+        "rcv": "LOXIQKHRCEDEMN5MKBQ32Y54J7H6JKBTEURQL4S65K44MTWN6UMTU2WNBM",
+        "amnt": "0",
+        "type": "pay"
+      }
+    }
+  ]
+}
+```
+
+## Signature Scheme
+
+Each transaction in the `paymentGroup` must be signed individually by each of the owners of the sender addresses in the group.
+
+In Algorand, signatures are based on the `Ed25519` signature scheme. The `signature` field in the `X-PAYMENT` header payload is a base64 encoded string of the signed transaction.
+
+### Algorand Addresses and Public Key relationship
+
+In Algorand, addresses are 58-character base32-encoded representations of public keys, with a sha512_256 checksum (last 4 bytes) appended as a suffix. This encoding ensures that addresses can be derived directly from public keys without requiring a pre-existing signature for operations like `ecRecover` to verify signatures.
+
+#### Example of encoding Algorand addresses:
+
+```ts
+	encodeAddress(publicKey: Buffer): string {
+		const keyHash: string = sha512_256.create().update(publicKey).hex()
+
+		// last 4 bytes of the hash
+		const checksum: string = keyHash.slice(-8)
+
+		return base32.encode(Encoder.ConcatArrays(publicKey, Buffer.from(checksum, "hex"))).slice(0, 58)
+	}
+```
+
+
+### Encoding
+
+#### Types
+
+Before encoding, all `txn` parameters must be converted into primitive types:
+
+- `gh`: The genesis hash of the network, which is a base64 encoded string should be decoded. 
+- `snd`: The sender address should be decoded into its public key bytes.
+- `rcv`: The receiver address should be decoded into it's public key bytes.
+- `xaid`: The asset ID should be converted into a uint64 integer.
+- `aamnt`: The amount to transfer should be converted into a uint64 integer.
+- `fee`: The fee for the transaction should be converted into a uint64 integer.
+- `fv`: The first valid round should be converted into a uint64 integer.
+- `lv`: The last valid round should be converted into a uint64 integer.
```

Comment:
What about the other fields:

- `type` (e.g., `pay`, `axfer`, etc.)
- `lx`, why might support leased transactions? This is a good feature to avoid transaction replay-ability
- `grp`, needed if we support groups
- `rekey`, this is very Algorand specific, but can’t we include it?
- `note`, 1k of arbitrary data as a notefield

Reply:
What about type-specific fields (e.g., for payment, asset transfer, asset freeze, etc.).

Reply:
We need to discuss if we should allow for everything in the context of x402.